projects/lights_control/inc/lights_blinker.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+#include "event_queue.h"
+#include "soft_timer.h"
+
+// state definitions for blinker
+typedef enum {
+  LIGHTS_BLINKER_STATE_OFF = 0,
+  LIGHTS_BLINKER_STATE_ON,
+  NUM_LIGHTS_BLINKER_STATES
+} LightsBlinkerState;
+
+typedef uint32_t LightsBlinkerDuration;
+
+// blinker members
+typedef struct LightsBlinker {
+  EventID event_id;
+  SoftTimerID timer_id;
+  LightsBlinkerState state;
+  LightsBlinkerDuration duration_us;
+} LightsBlinker;
+
+// initializes the blinker
+StatusCode lights_blinker_init(LightsBlinker *);
```

Comment:
Missing parameter names?

Reply:
It's usually standard to have the init function take in parameters that it populates provided storage with if there aren't too many parameters.

Reply:
ah true, but in this case the blinker doesn't require any specific parameters. all it does is that it initializes the `timer_id` to an invalid timer id to indicate that this timer is not currently being used. 
C by default may initialize timer_id to 0, which could be problematic as that's a valid timer id.

Reply:
Where does the event ID come from then? Why not just store the time in init? Shouldn't it always have the same period?

I'm a little confused about the purpose of this module - all of our blinking lights should be synced, but this seems to create a bunch of independent blinkers.